torch/nn/modules/module.py

```diff
@@ -251,7 +252,12 @@ def compute_should_use_set_data(tensor, tensor_applied):
 
         for key, buf in self._buffers.items():
             if buf is not None:
-                self._buffers[key] = fn(buf)
+                buf_applied = fn(buf)
```

Comment:
Should this be run under a `no_grad` block?

Reply:
You are right! Buffers almost never require grads, but if they do, this will be an error.